pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/NamespacesImpl.java

```diff
@@ -146,6 +147,56 @@ public void failed(Throwable throwable) {
         return future;
     }
 
+    @Override
+    public List<BundleStats> getAllBundleStats(String namespace) throws PulsarAdminException {
+        return sync(() -> getAllBundleStatsAsync(namespace));
+    }
+
+    @Override
+    public CompletableFuture<List<BundleStats>> getAllBundleStatsAsync(String namespace) {
+        NamespaceName ns = NamespaceName.get(namespace);
+        WebTarget path = namespacePath(ns, "bundleStats");
+        final CompletableFuture<List<BundleStats>> future = new CompletableFuture<>();
+        asyncGetRequest(path,
```

Comment:
Will there be a lookup here? All bundles of a namespace will be on different brokers.

Reply:
No, the API will only return the bundles owned by the broker which this http request sent to.

Reply:
Ok, Is the interface name ambiguity?

Reply:
> Ok, Is the interface name ambiguity?

Now, The meaning of the interface maybe misunderstood: `Get the status of all bundles under the namespace`

Reply:
> Ok, Is the interface name ambiguity?
Now, The meaning of the interface maybe misunderstood: `Get the status of all bundles under the namespace`

Reply:
I made a similar point in another comment. Since the admin cli doesn't parameterize which broker is targeted, I think this endpoint will lead to unexpected results when there is more than one broker. I think this might even be true when looking up a singe bundle's stats, unless there is a lookup query somewhere that I'm missing.